src/contexts/ServerContext.js

```diff
@@ -528,11 +531,25 @@ export const ServerProvider = ({ children }) => {
 
 
     const loadData = () => {
+        // not including servers key because we don't need to set default load amount here for it
+        const pageKeys = ["groupsPerPageKey", "resourcesPageKey", "usersPageKey"];
+
+        for (let key of pageKeys) {
+            if (!localStorage.getItem(environment[key])) {
+                localStorage.setItem(environment[key], environment.defaultItemsPerPage);
+            }
+        }
+
+        const groupsPerPage = parseInt(localStorage.getItem(environment[pageKeys[0]]), 10);
+        const rescPerPage = parseInt(localStorage.getItem(environment[pageKeys[1]]), 10);
+        const usersPerPage = parseInt(localStorage.getItem(environment[pageKeys[2]]), 10);
+        
+
         !isLoadingZones && loadZones();
         !isLoadingZoneContext && loadServers();
-        !isLoadingUserContext && loadUsers(0, 10, '', 'asc', 'USER_NAME');
-        !isLoadingGroupContext && loadGroups(0, 10, '', 'asc', 'USER_NAME');
-        !isLoadingRescContext && loadResources(0, 0, '', 'asc', 'RESC_NAME');
+        !isLoadingUserContext && loadUsers(0, usersPerPage, '', 'asc', 'USER_NAME');
+        !isLoadingGroupContext && loadGroups(0, groupsPerPage, '', 'asc', 'USER_NAME');
+        !isLoadingRescContext && loadResources(0, rescPerPage, '', 'asc', 'RESC_NAME');
```

Comment:
let's keep these in the same order as above.   no functional change, just aesthetic.

Reply:
yep, i like the consistency better if its the same order as well. changed to

```js
!isLoadingGroupContext && loadGroups(0, groupsPerPage, '', 'asc', 'USER_NAME');
!isLoadingRescContext && loadResources(0, rescPerPage, '', 'asc', 'RESC_NAME');
!isLoadingUserContext && loadUsers(0, usersPerPage, '', 'asc', 'USER_NAME');
```

Reply:
perfect